src/compiler/typeTests.ts

```diff
@@ -0,0 +1,18 @@
+/* @internal */
+namespace ts {

```

Comment:
I'm new-ish to the codebase, so take this with a grain of salt, but it seems like the convention across the board is to inline these checks and cast, rather than to use type guards. Maybe for performance reasons?

Seems a little suspicious that these pretty basic type guard functions don't exist already, in that they are convenient, so surely they would exist already if there wasn't a good reason for them not to exist.

Reply:
Hi, thanks for the review.

Yes, it is a bit suspicious that such basic things don't exist - if there's a good reason for this, I'll inline the checks as you suggested.

I generally tried to improve the code quality where I could, including this - it seemed really obvious that this could help readability and correctness. I am, however, a first time contributor, so I don't know the reasoning behind the design choices...

Reply:
Obviously, since we don't have guards like this, our prevailing style is to do the check and cast as-needed. *Most* operations don't need these `Type` subtypes explicitly. (Because the lines between them are blurry at times.) It's better to just match the existing style, at least for now.